l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -472,4 +497,93 @@ contract ExecutorFacet is Base, IExecutor {
     function _setBit(uint256 _bitMap, uint8 _index) internal pure returns (uint256) {
         return _bitMap | (1 << _index);
     }
+
+    /// @notice Calls the point evaluation precompile and verifies the output
+    /// Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof.
+    /// Also verify that the provided commitment matches the provided versioned_hash.
+    /// 
+    function _pointEvaluationPrecompile(
+        bytes32 _versionedHash,
+        bytes32 _openingPoint,
+        bytes calldata _openingValueCommitmentProof
+    ) internal view {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _openingPoint, _openingValueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.staticcall(precompileInput);
+
+        // We verify that the point evaluation precompile call was successful by testing the latter 32 bytes of the
+        // response is equal to BLS_MODULUS as defined in https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile
+        require(success, "failed to call point evaluation precompile");
+        (, uint256 result) = abi.decode(data, (uint256, uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    /// Verifies that the blobs contain the correct data by calling the point evaluation precompile. For the precompile we need:
+    /// versioned hash || opening point || opening value || commitment || proof
+    /// the _pubdataCommitments will contain the last 4 values, the versioned hash is pulled from the BLOBHASH opcode
+    /// pubdataCommitments is a list of: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || proof (48 bytes)) = 144 bytes
+    function _verifyBlobInformation(
+        bytes calldata _pubdataCommitments,
+        bytes32[MAX_NUMBER_OF_BLOBS] memory _blobHashes
+    ) internal view returns (bytes32[] memory blobCommitments) {
+        uint256 versionedHashIndex = 0;
+
+        require(_pubdataCommitments.length > 0, "pl");
+        require(
+            _pubdataCommitments.length <= PUBDATA_COMMITMENT_SIZE * MAX_NUMBER_OF_BLOBS &&
+                _pubdataCommitments.length % PUBDATA_COMMITMENT_SIZE == 0,
+            "bs"
+        );
+        blobCommitments = new bytes32[](MAX_NUMBER_OF_BLOBS);
+        bytes32 versionedHash;
```

Comment:
nit: since we don't need to carry this variable's value between the `for` loops it is nicer to create it only when needed to be used (i.e. inside the for loop and right after it)